start_geocint.sh

```diff
@@ -48,24 +52,24 @@ trap 'cleanup' EXIT
 
 # Update runner if updating is true in config 
 if [ "$UPDATE_RUNNER" = "true" ]; then
-  cd ~/geocint-runner; git pull --rebase --autostash || { git stash && git pull && echo 'git rebase autostash geocint-runner failed, stash and pull executed' | python scripts/slack_message.py $SLACK_CHANNEL "$SLACK_BOT_NAME" $SLACK_BOT_EMOJI; }
+  cd ${GEOCINT_WORK_DIRECTORY}/geocint-runner; git pull --rebase --autostash || { git stash && git pull && echo 'git rebase autostash geocint-runner failed, stash and pull executed' | python scripts/slack_message.py $SLACK_CHANNEL "$SLACK_BOT_NAME" $SLACK_BOT_EMOJI; }
 fi
 
 # Update osm logic if updating is true in config
 if [ "$UPDATE_OSM_LOGIC" = "true" ]; then
-  cd ~/geocint-openstreetmap; git pull --rebase --autostash || { git stash && git pull && echo 'git rebase autostash geocint-openstreetmap failed, stash and pull executed' | python scripts/slack_message.py $SLACK_CHANNEL "$SLACK_BOT_NAME" $SLACK_BOT_EMOJI; }
+  cd ${GEOCINT_WORK_DIRECTORY}/geocint-openstreetmap; git pull --rebase --autostash || { git stash && git pull && echo 'git rebase autostash geocint-openstreetmap failed, stash and pull executed' | python scripts/slack_message.py $SLACK_CHANNEL "$SLACK_BOT_NAME" $SLACK_BOT_EMOJI; }
 fi
 
 # Update private repo if updating is true in config
 if [ "$UPDATE_PRIVATE" = "true" ]; then
-  cd ~/$PRIVATE_REPO_NAME; git pull --rebase --autostash || { git stash && git pull && echo 'git rebase autostash $PRIVATE_REPO_NAME failed, stash and pull executed' | python scripts/slack_message.py $SLACK_CHANNEL "$SLACK_BOT_NAME" $SLACK_BOT_EMOJI; }
+  cd ${GEOCINT_WORK_DIRECTORY}/$PRIVATE_REPO_NAME; git pull --rebase --autostash || { git stash && git pull && echo 'git rebase autostash $PRIVATE_REPO_NAME failed, stash and pull executed' | python scripts/slack_message.py $SLACK_CHANNEL "$SLACK_BOT_NAME" $SLACK_BOT_EMOJI; }
 fi
 
 # Remove all files and folders except data, db, deploy and logs from the general folder 
-find ~/$GENERAL_FOLDER/. -maxdepth 1 -type d -not -name  "d*" -not -name '*.*' -not -name  "logs" -not -name  "reports" | xargs rm -rf
-find ~/$GENERAL_FOLDER/ -maxdepth 1 -type f -not -name "make_profile.db" -delete
+find ${GEOCINT_WORK_DIRECTORY}/$GENERAL_FOLDER/. -maxdepth 1 -type d -not -name  "d*" -not -name '*.*' -not -name  "logs" -not -name  "reports" -not -name  "installation" | xargs rm -rf
+find ${GEOCINT_WORK_DIRECTORY}/$GENERAL_FOLDER/ -maxdepth 1 -type f -not -name "make_profile.db" -delete
```

Comment:
ok, if there will be something, for example, mapaction specific that lying somewhere not inside data, db, deploy directories that also doesn't need to be deleted before next run how it should be configured?

Reply:
we can replace it with regex:
find ${GEOCINT_WORK_DIRECTORY}/$GENERAL_FOLDER/. -maxdepth 1 -type d -regex ".*/d.[^/]*\|.*/logs[^.]\|.*/reports[^.]\|.*installation[^.]" 

and add this regex as a variable to config @kalenikaliaksandr

Reply:
also if ${GEOCINT_WORK_DIRECTORY} and $GENERAL_FOLDER are empty this will remove all files in the filesystem

Reply:
Replaced with variables 
KEEP_FOLDERS_REGEX='.*/\|.*/data[^.]*\|.*/db[^.]*\|.*/deploy[^.]*\|.*/logs[^.]*\|.*/reports[^.]*\|.*/installation[^.]*'
KEEP_FILES_REGEX=".*/make_profile\.db[^.]*"

Reply:
added empty variables check to make execution safe